server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -954,11 +964,16 @@ public static String findV2Template(Metadata metadata, String indexName, boolean
             .map(ComponentTemplate::template)
             .map(Template::mappings)
             .filter(Objects::nonNull)
-            .collect(Collectors.toList());
+            .collect(Collectors.toCollection(LinkedList::new));
```

Comment:
because we insert a mapping on the first position below (adding the default `@timestamp`). That takes linear time with `ArrayList` as opposed to constant time with the `LinkedList`. We also don't random access the mappings list but always iterate it on, so it seems like the `LinkedList` is a better fit for the current use cases.